File: vscode-extension/src/protocol/NodeMetadata.ts
/*------------------------------------------------------------------------------------------------
 *  Copyright (c) 2022 ICSharpCode
 *  Licensed under the MIT License. See LICENSE.TXT in the project root for license information.
 *-----------------------------------------------------------------------------------------------*/

import { NodeType } from "./NodeType";

export default interface NodeMetadata {
  assemblyPath: string;
  type: NodeType;
  name: string;
  symbolToken: number;
  parentSymbolToken: number;
  subType?: string;
}

File: backend/ILSpyX.Backend/Model/NodeMetadata.cs
// Copyright (c) 2022 ICSharpCode
// Licensed under the MIT license. See the LICENSE file in the project

namespace ILSpyX.Backend.Model;

public record NodeMetadata
{
    public required string AssemblyPath { get; init; }
    public required NodeType Type { get; init; }
    public required string Name { get; init; }
    public string? BundledAssemblyName { get; init; }
    public int SymbolToken { get; init; } // = 0
    public int ParentSymbolToken { get; init; } // = 0
    public bool IsDecompilable { get; init; } // = false
    public string? SubType { get; init; }
}